R/epidist.R

```diff
@@ -814,19 +812,147 @@ parameters.epidist <- function(x, ...) {
     params <- unlist(x$prob_dist$parameters)
   } else if (inherits(x$prob_dist, "distribution")) {
     params <- unlist(distributional::parameters(x$prob_dist))
+
     # if dist is truncated clean names
-    if (identical(stats::family(x$prob_dist), "truncated")) {
+    if (is_truncated(x)) {
       names(params) <- gsub(
         pattern = "dist.",
         replacement = "",
         x = names(params),
         fixed = TRUE
       )
     }
+
+    # convert to meanlog and sdlog names
+    class(params) <- family(x)
+    params <- clean_epidist_params(prob_dist_params = params)
+
   } else {
     stop("Distribution in `epidist` not recognised.")
   }
 
   # return parameters
   params
 }
+
+#' Family method for the `epidist` class
+#'
+#' @description The `family()` function is used to extract the distribution
+#' names from objects from {distributional} and {distcrete}. This method
+#' provides the same interface for `<epidist>` objects to give consistent
+#' output irrespective of the distribution class.
+#'
+#' @param object An `epidist` object
+#' @inheritParams stats::family
+#'
+#' @return A character string with the name of the distribution
+#' @importFrom stats family
+#' @export
+#'
+#' @examples
+#' # example with continuous distribution
+#' edist <- epidist(
+#'   disease = "ebola",
+#'   epi_dist = "incubation_period",
+#'   prob_distribution = "gamma",
+#'   prob_distribution_params = c(shape = 1, scale = 1)
+#' )
+#' family(edist)
+#'
+#' # example with discretised distribution
+#' edist <- epidist(
+#'   disease = "ebola",
+#'   epi_dist = "incubation_period",
+#'   prob_distribution = "lnorm",
+#'   prob_distribution_params = c(meanlog = 1, sdlog = 1),
+#'   discretise = TRUE
+#' )
+#' family(edist)
+family.epidist <- function(object, ...) {
+
+  if (inherits(object$prob_dist, "distcrete")) {
+    prob_dist <- object$prob_dist$name
+  } else {
+    prob_dist <- stats::family(object$prob_dist)
+
+    if (identical(prob_dist, "truncated")) {
```

Comment:
```suggestion
    if (is_truncated(object)) {
```

If I understood correctly